### PR TITLE
pybind/mgr/balancer/module.py: add max/min info in stats_by_root

### PR DESCRIPTION
**Pull request by yanghonggang**
$ ceph balancer eval-verbose hahapool
...
count_by_pool {'hahapool': {'objects': {0: 1L, 1: 1L, 2: 0L, 3: 0L, 4: 1L}, 'bytes': {0: 0L, 1: 0L, 2: 0L, 3: 0L, 4: 0L}, 'pgs': {0: 35, 1: 42, 2: 23, 3: 30, 4: 26}}}
...
stats_by_root {'default': {'objects': {'min': 0.0, 'max': 1.0, 'score': 0.2970089549436925, 'stddev': 0.5477225575051662, 'avg': 0.6, 'sum_weight': 0.6000000000000001}, 'bytes': {'min': 0, 'max': 0, 'score': 0, 'stddev': 0, 'avg': 0, 'sum_weight': 0}, 'pgs': {'min': 23.0, 'max': 42.0, 'score': 0.07354228523075643, 'stddev': 7.52994023880668, 'avg': 31.2, 'sum_weight': 0.4}}}

max/min can give us an intuitive feel of the pg/bytes/object's distribution.
In addition, sometimes we really want to know which osd hold the most
number of pgs/bytes/objects for specified pool.

Signed-off-by: Yang Honggang <yanghonggang@umcloud.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by yanghonggang** [comment]
@xiexingguo  @tchaikov

File: src/pybind/mgr/balancer/module.py [hunk 1]
```diff
@@ -189,6 +191,8 @@ def calc_stats(self, count, target, total):
             stddev = math.sqrt(dev / float(max(num - 1, 1)))
             score = score / max(sum_weight, 1)
             r[t] = {
+                'max': max(count[t].iteritems(), key=lambda x:x[1])[1],
```
Comment (xiexingguo) [inline src/pybind/mgr/balancer/module.py]: Can we fold the min/max calculation into the `for k, v in six.iteritems(count[t]):` loop, which looks more straightforward and efficient..

Comment (xiexingguo) [reply src/pybind/mgr/balancer/module.py]: or max(count[t].itervalues()) should suffice...

Comment (yanghonggang) [reply src/pybind/mgr/balancer/module.py]: > or max(count[t].itervalues()) should suffice...

done @xiexingguo

File: src/pybind/mgr/balancer/module.py [hunk 2]
```diff
@@ -189,6 +191,8 @@ def calc_stats(self, count, target, total):
             stddev = math.sqrt(dev / float(max(num - 1, 1)))
             score = score / max(sum_weight, 1)
             r[t] = {
+                'max': max(count[t].itervalues()),
+                'min': min(count[t].itervalues()),
```
Comment (tchaikov) [inline src/pybind/mgr/balancer/module.py]: `itervalues()` is not compatible with py3.

Comment (xiexingguo) [reply src/pybind/mgr/balancer/module.py]: @yanghonggang mind switching to  `values()`? then this should be good to go

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-09-21_17:17:30-rados-wip-kefu-testing-2019-09-20-1944-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-09-21_17:17:30-rados-wip-kefu-testing-2019-09-20-1944-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-09-22_03:04:52-rados-wip-kefu-testing-2019-09-20-1945-distro-basic-mira/

failures tracked by

- https://tracker.ceph.com/issues/41744
- https://tracker.ceph.com/issues/41946

**Pull request merged**